sunpy/map/mapbase.py

```diff
@@ -2739,6 +2740,14 @@ def reproject_to(self, target_wcs, *, algorithm='interpolation', return_footprin
         outmap = GenericMap(output_array, target_wcs.to_header(),
                             plot_settings=self.plot_settings)
 
+        # Check rsun mismatch
+        rsun_map = self.meta["rsun_ref"]
+        rsun_target = outmap.meta["rsun_ref"]
+        rsun_diff = abs(rsun_map - rsun_target)
+        if rsun_diff > 200000:
```

Comment:
Can you suggest why you picked this value?

Reply:
```python
from sunpy.map import Map
import numpy as np

aiamap = Map("https://git.ias.u-psud.fr/adolliou/report_issue_github/-/raw/main/data/sunpy_reprojection_issue/aia.lev1.171.1497617052.2022-03-17T00_22_03.342Z.image_lev1.fits?ref_type=heads")
hrimap = Map("https://git.ias.u-psud.fr/adolliou/report_issue_github/-/raw/main/data/sunpy_reprojection_issue/solo_L2_eui-hrieuvopn-image_20220317T001800215_V02.fits?ref_type=heads")

aiamap_rsun_ref_val = aiamap.meta["rsun_ref"]
hrimap_rsun_ref_val = hrimap.meta["rsun_ref"]

# Looping through different rsun_ref differences to evaluate has_nan behavior
for i in range(101790, 101800, 1):
    # Modifying aiamap rsun_ref based on the iteration
    aiamap.meta["rsun_ref"] = aiamap_rsun_ref_val - i

    # Calculate the difference in rsun_ref values between the maps
    rsun_diff = abs(aiamap.meta["rsun_ref"] - hrimap_rsun_ref_val)

    # Reprojecting AIA map to HRI map and checking for NaN values
    aia_reprojected = aiamap.reproject_to(hrimap.wcs, algorithm="interpolation")
    has_nan = np.isnan(aia_reprojected.data).any()

    # Print the rsun_diff and has_nan status
    print(f"rsun_diff: {rsun_diff}, has_nan: {has_nan}")
    # aia_reprojected.peek()
```

```python
Output:
rsun_diff: 198210.0, has_nan: True
rsun_diff: 198209.0, has_nan: True
rsun_diff: 198208.0, has_nan: True
rsun_diff: 198207.0, has_nan: True
rsun_diff: 198206.0, has_nan: True
rsun_diff: 198205.0, has_nan: False
rsun_diff: 198204.0, has_nan: False
rsun_diff: 198203.0, has_nan: False
rsun_diff: 198202.0, has_nan: False
rsun_diff: 198201.0, has_nan: False
```

As `rsun_diff` decreased from 198210 to 198201, the `has_nan` transitioned from `True` (indicating data loss) to `False` (no data loss). Notably, at the `rsun_diff` value of 198205, the `has_nan` status changed to `False`, suggesting no data loss beyond this point. Therefore, `rsun_diff` values greater than 198205 consistently caused data loss during reprojection.

Based on the above evaluation I have selected new a threshold of 198205 which I believe is precise compared to 20000.

Reply:
Hmm, the original issue suggested raising it if the values are different at all. 

@ayshih what are your thoughts on this?

Reply:
Rigorously, there is no special threshold at which a difference in rsun becomes problematic.  @Mubin17: The hidden assumption in your investigation is the pixel size of the original map.  Once the difference in rsun is smaller than the pixel size, it's possible to get through a reprojection without losing any data.

While a constant threshold for rsun difference would not work, there is the intriguing possibility of implementing a heuristic threshold derived from the pixel size of the original map.  Barring that, the safest thing to do for the time being is to raise the warning for any difference in rsun, no matter how small.

Reply:
@ayshih @nabobalis Thanks for the clarification. I've implemented your feedback, raising the warning for any difference in `rsun`, regardless of its magnitude.